crates/burrego/src/opa/builtins/glob.rs

```diff
@@ -0,0 +1,46 @@
+use anyhow::{anyhow, Result};
+
+pub fn quote_meta(args: &[serde_json::Value]) -> Result<serde_json::Value> {
+    if args.len() != 1 {
+        return Err(anyhow!("glob.quote_meta: wrong number of arguments"));
+    }
+
+    let input = args[0]
+        .as_str()
+        .ok_or_else(|| anyhow!("glob.quote_meta: 1st parameter is not a string"))?;
+
+    serde_json::to_value(escape(input))
+        .map_err(|e| anyhow!("glob.quote_meta: cannot convert value into JSON: {:?}", e))
+}
+
+fn escape(s: &str) -> String {
+    let mut escaped = String::new();
+    for c in s.chars() {
+        match c {
+            '?' | '*' | '[' | ']' => {
```

Comment:
Looking at the `glob.match` documentation I think we might have to handle also: `:`, `{` and `}`.

I haven't looked at the source code of opa though...

Reply:
The go implementation uses https://github.com/gobwas/glob. I'm going to expand the quote so we match all special characters.

Reply:
Their implementation check the glob special chars and escapes by prepending a `\`: https://github.com/open-policy-agent/opa/blob/581ba9c456be551b584759ae91f99ae6ec6b347b/vendor/github.com/gobwas/glob/glob.go#L68-L77.

The list of special characters is determined by: https://github.com/open-policy-agent/opa/blob/581ba9c456be551b584759ae91f99ae6ec6b347b/vendor/github.com/gobwas/glob/syntax/lexer/lexer.go#L23-L35.

I'm just copy-catting this behavior here, updated the PR.